### ISSUE TITLE
UX: Mouse back and forward (mouse4 and mouse5) buttons do not work

### ISSUE THREAD

**Issue by PolarPayne** [SerenityOS/serenity#13627]
I would expect the mouse back and forward buttons to go backwards and forwards when it makes sense. For example, in the browser and file manager they should map to the "Go Back" and "Go Forward" actions. This should probably be somewhat general in the UI library so that it would work more less "out of the box correctly".

**Comment by PolarPayne** [comment]
Neat. Hope it gets finished and merged.

**Comment by GMTA** [comment]
> PR #12291 is a solution to this.

That looks like a LibGUI-only change, we would need to support the mouse forward and backwards buttons in the device as well. I.e.:
https://github.com/SerenityOS/serenity/blob/00099a8adef8be9633e17405ccd66faa3cfe4c1f/Kernel/Devices/VMWareBackdoor.cpp#L236-L241

**Comment by AtkinsSJ** [comment]
Huh... I was sure someone already added the hardware support a while ago, but I can't find any PRs for it.

**Comment by geordiemhall** [comment]
Yeah it was working for me when testing in QEMU on Windows via WSL2, but maybe there are other hardware/codepaths that still lack support?

**Comment by GMTA** [comment]
So PS/2 support is there:
https://github.com/SerenityOS/serenity/blob/66170ff632c58f7c8689e08eda218037b63632c3/Kernel/Devices/HID/PS2MouseDevice.cpp#L127-L132

But when the VMWare backdoor is active, this code path is not taken for the mouse.

**Comment by supercomputer7** [comment]
> So PS/2 support is there:
> 
> https://github.com/SerenityOS/serenity/blob/66170ff632c58f7c8689e08eda218037b63632c3/Kernel/Devices/HID/PS2MouseDevice.cpp#L127-L132
> 
> But when the VMWare backdoor is active, this code path is not taken for the mouse.

When using the vmmouse feature, the mouse only supports 3 buttons and not 5.
I verified this with the QEMU code too.
